contracts/src/v0.8/ccip/FeeQuoter.sol

```diff
@@ -848,12 +852,73 @@ contract FeeQuoter is AuthorizedCallers, IFeeQuoter, ITypeAndVersion, IReceiver,
   /// @param chainFamilySelector Tag to identify the target family.
   /// @param destAddress Dest address to validate.
   /// @dev precondition - assumes the family tag is correct and validated.
+  /// @dev Since Solana addresses are parsed as bytes32, and no other form of validation occurs, no explicit
+  /// call is needed to a library function for address validation.
   function _validateDestFamilyAddress(bytes4 chainFamilySelector, bytes memory destAddress) internal pure {
     if (chainFamilySelector == Internal.CHAIN_FAMILY_SELECTOR_EVM) {
       Internal._validateEVMAddress(destAddress);
     }
   }
 
+  function _parseGasLimitFromExtraArgBytes(
+    bytes calldata extraArgs,
+    DestChainConfig memory destChainConfig
+  ) internal pure returns (uint256 gasLimit) {
+    if (destChainConfig.chainFamilySelector == Internal.CHAIN_FAMILY_SELECTOR_EVM) {
+      gasLimit = _parseEVMExtraArgsFromBytes(extraArgs, destChainConfig).gasLimit;
+    } else if (destChainConfig.chainFamilySelector == Internal.CHAIN_FAMILY_SELECTOR_SOL) {
+      // If extra args are empty, generate default values.
+      gasLimit = _parseSolExtraArgsFromBytes(extraArgs, destChainConfig, 0).computeUnits;
+    } else {
+      gasLimit = destChainConfig.defaultTxGasLimit;
+    }
+
+    return gasLimit;
+  }
+
+  /// @notice Parse and validate the Solana specific Extra Args Bytes.
+  /// @param messageDataLengthBytes The length of the arbitrary message data. If this is non-zero, then an extraArgs
+  /// MUST be provided otherwise an error will be returned, due to requirements in how Solana accounts operate.
+  function _parseSolExtraArgsFromBytes(
+    bytes calldata extraArgs,
+    DestChainConfig memory destChainConfig,
+    uint256 messageDataLengthBytes
+  ) internal pure returns (Client.SolExtraArgsV1 memory) {
+    Client.SolExtraArgsV1 memory SolExtraArgs =
+      _parseUnvalidatedSolExtraArgsFromBytes(extraArgs, destChainConfig.defaultTxGasLimit);
+
+    // If the message data length is non-zero, then accounts extra args must be provided.
+    if (messageDataLengthBytes != 0 && SolExtraArgs.accounts.length == 0) revert SolExtraArgsMustBeProvided();
```

Comment:
We need to remove this validation as well, as we need to support executing a message with data.lenght == 0
```suggestion
```